dotCMS/src/main/java/com/dotcms/contenttype/model/type/BaseContentType.java

```diff
@@ -2,19 +2,39 @@
 
 import com.fasterxml.jackson.annotation.JsonValue;
 
+/**
+ * Provides the different Content Types that can be used inside dotCMS. A
+ * Content Type allows users to shape content in the application, i.e., the
+ * fields used by authors to add content or information to their sites.
+ * 
+ * @author Will Ezell
+ * @version 4.1.0
+ * @since Oct 17, 2016
+ *
+ */
 public enum BaseContentType {
+
 	ANY(0, ContentType.class),
 	CONTENT(1, SimpleContentType.class),
 	WIDGET(2, WidgetContentType.class),
-	FORM(3,FormContentType.class),
+	FORM(3, FormContentType.class),
 	FILEASSET(4, FileAssetContentType.class),
 	HTMLPAGE(5, PageContentType.class),
-	PERSONA(6,PersonaContentType.class);
+	PERSONA(6, PersonaContentType.class),
+	VANITY_URL(7, VanityUrlContentType.class);
 
 	final int type;
-	Class immutableClass;
-	
-	BaseContentType(int type, Class clazz) {
+	Class<? extends ContentType> immutableClass;
+
+	/**
+	 * Enum's constructor.
+	 * 
+	 * @param type
+	 *            - The integer representation of the Content Type.
+	 * @param clazz
+	 *            - The class of the specific Content Type.
+	 */
+	BaseContentType(int type, Class<? extends ContentType> clazz) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "BaseContentType" constructor. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)